ibstore/_store.py

```diff
@@ -454,25 +522,47 @@ def optimize_mm(
             chunksize=chunksize,
         )
 
-        for result in _minimized_blob:
-            inchi_key = result.inchi_key
-            molecule_id = self.get_molecule_id_by_inchi_key(inchi_key)
-            self.store_conformer(
-                MMConformerRecord(
+        inchi_to_id = {
+            inchi_key: id
+            for (id, inchi_key) in reversed(
+                db.db.query(
+                    DBMoleculeRecord.id,
+                    DBMoleculeRecord.inchi_key,
+                ).all()
+            )
+        }
+
+        with self._get_session() as db:
+            # from _mm_conformer_already_exists
+            seen = set(
+                db.db.query(
+                    DBMMConformerRecord.qcarchive_id,
+                )
+            )
+            for result in _minimized_blob:
```

Comment:
I can't highlight the diff cleanly, but could the `in seen` check happen immediately after getting a `result` from the iterator?

```suggestion
            for result in _minimized_blob:
            if result.qcarchive_id in seen:
                continue
```

This is not likely to be the bottleneck, but it looks safe to short-circuit as early as possible